src/core/color-strategies.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import type Highcharts from "highcharts";
+
+const DEFAULT_COLOR = "black";
+
+export interface ColorStrategy {
+  getColor(series: Highcharts.Series, chart: Highcharts.Chart): string;
+}
+
+export class DefaultColorStrategy implements ColorStrategy {
+  // Highcharts supports color objects to represent gradients and more, but we only support string
+  // colors in our markers. In case a non-string color is defined, we use black color as as fallback.
+  getColor(series: Highcharts.Series): string {
+    return typeof series?.color === "string" ? series.color : DEFAULT_COLOR;
+  }
+}
+
+export class SolidGaugeColorStrategy implements ColorStrategy {
+  getColor(series: Highcharts.Series, chart: Highcharts.Chart): string {
+    const seriesYAxis = series.yAxis || (chart.yAxis && chart.yAxis[0]);
+    if (!seriesYAxis) {
+      return DEFAULT_COLOR;
+    }
+
+    const dataValue = series.data?.[0]?.y;
+    return this.calculateColorFromStops(seriesYAxis, dataValue);
+  }
+
+  /**
+   * Calculates the color for a value based on the y-axis color stops configuration.
+   *
+   * @param yAxis - The Highcharts axis object containing the stops configuration
+   * @param value - The data value to calculate the color for
+   * @returns The color string from the matching stop, or the default color if:
+   *          - No stops are defined
+   *          - No value is provided (returns first stop color)
+   *          - Value doesn't match any stops
+   *          - Color in stop is not a string
+   *          - Any error occurs during calculation
+   *
+   * The matching logic works as follows:
+   * 1. The input value is normalized to a scale of 0-1 based on the axis min/max
+   * 2. Stops are checked from highest position to lowest (reverse order)
+   * 3. The first stop where normalized value >= stop position is used
+   * 4. For example, with stops at [0, 'red'], [0.5, 'yellow'], [0.8, 'green']:
+   *    - Value 0.3 matches position 0 -> returns 'red'
+   *    - Value 0.7 matches position 0.5 -> returns 'yellow'
+   *    - Value 0.9 matches position 0.8 -> returns 'green'
+   * 5. If no matching stop is found, returns color from first stop
+   */
+  private calculateColorFromStops(yAxis: Highcharts.Axis, value?: number): string {
+    try {
+      const stops =
+        (yAxis.options as Highcharts.YAxisOptions)?.stops || (yAxis.userOptions as Highcharts.YAxisOptions)?.stops;
+      const min = yAxis.min ?? 0;
+      const max = yAxis.max ?? 100;
+
+      if (!stops?.length) {
+        return DEFAULT_COLOR;
+      }
+      if (value === null || value === undefined) {
```

Comment:
In what case the value can be undefined?

If the value is null - why are we taking the first stop for it? Should we do some fallback? E.g. if the explicit series colour are accepted, we can take that instead.